bootstrap-core/src/main/java/de/agilecoders/wicket/core/markup/html/bootstrap/dialog/Alert.java

```diff
@@ -85,21 +87,50 @@ public Alert(String id, IModel<String> message, IModel<String> header) {
 
         type = Type.Info;
         dismissable = false;
+        this.header = header;
+    }
 
-        this.inlineHeader = new Label("inline", header);
-        this.blockHeader = new Label("block", header);
-        this.message = createMessage("message", getModel());
-        this.closeButton = new WebMarkupContainer("close") {
-            private static final long serialVersionUID = 1L;
+    private Label getInlineHeader() {
+        if (inlineHeader == null) {
+            inlineHeader = new Label("inline", header);
+        }
+        return inlineHeader;
+    }
 
-            @Override
-            protected void onConfigure() {
-                super.onConfigure();
-                setVisible(dismissable);
-            }
-        };
+    private Label getBlockHeader() {
+        if (blockHeader == null) {
+            blockHeader = new Label("block", header);
+        }
+        return blockHeader;
+    }
 
-        add(this.inlineHeader, this.blockHeader, this.message, this.closeButton);
+    protected Component getMessage() {
```

Comment:
Does this need to be `protected` ?
The idea is to not call overridable methods in the constructor because their bodies may assume that the current instance is fully initialized.

Reply:
Otherwise NotificationAlert get broken .... :(

Reply:
OK. Please add `final` to it then